pkg/blockstorage/azure/azuredisk.go

```diff
@@ -383,34 +387,73 @@ func (s *AdStorage) VolumeParse(ctx context.Context, volume interface{}) (*block
 	if z := vol.Zones; len(z) > 0 {
 		az = az + "-" + *(z[0])
 	}
+	volumeType := ""
+	if vol.SKU != nil &&
+		vol.SKU.Name != nil {
+		volumeType = string(*vol.SKU.Name)
+	} else {
+		return nil, errors.New("Volume type is not available")
```

Comment:
[ni] How come we return errors for some `nil` pointer checks but not the other? 

This is not a blocker but rather a question on consistently. And my guess is that lac of critical data results in an error while lack of non-critical data does not. But I'd like to make sure that is the case. And we should document in the function comment what is considered critical data. It would help w quick debugging in the future.